test/unit/bucket/test_bucket.py

```diff
@@ -1402,6 +1402,20 @@ def test_upload_local_file_retention(self):
             self.assertEqual(retention, file_info.file_retention)
             self.assertEqual(LegalHold.ON, file_info.legal_hold)
 
+    def test_upload_bytes_cache_control(self):
+        data = b'hello world'
+        file_info = self.bucket.upload_bytes(data, 'file1', cache_control='max-age=3600')
+        self.assertTrue(isinstance(file_info, VFileVersionInfo))
+        self._check_file_contents('file1', data)
+        self.assertEqual(file_info.cache_control, 'max-age=3600')
+
+    def test_upload_file_cache_control(self):
+        data = b'hello world'
+        file_info = self.bucket.upload_bytes(data, 'file1', cache_control='max-age=3600')
+        self.assertTrue(isinstance(file_info, VFileVersionInfo))
+        self._check_file_contents('file1', data)
+        self.assertEqual(file_info.cache_control, 'max-age=3600')
+
```

Comment:
these tests are nearly identical - please refactor them in such a way that it doesn't cause so much code duplication